src/axom/primal/operators/detail/in_curved_polygon_impl.hpp

```diff
@@ -183,28 +183,27 @@ double adaptive_winding_number(const Point<T, 2>& q,
   if(ord <= 0) return 0.0;  // Catch degenerate cases
 
   // Use linearity as base case for recursion
-  if(c.isLinear(EPS))
-    return 0.0 - linear_winding_number(q, c[0], c[ord], edge_tol);
+  if(c.isLinear(EPS)) return linear_winding_number(q, c[0], c[ord], edge_tol);
 
   Polygon<T, 2> controlPolygon(c.getControlPoints());
 
-  // If q is outside the control polygon, for an open Bezier curve, the winding
-  //  number for the shape connected at the endpoints with straight lines is zero.
-  //  We then subtract the contribution of this line segment.
-  if(!in_polygon(q, controlPolygon, true, false, EPS))
-    return 0.0 - linear_winding_number(q, c[0], c[ord], edge_tol);
-
   // Check if our new curve is convex.
   //  If so, all subcurves will be convex as well
   if(!isConvexControlPolygon)
     isConvexControlPolygon = is_convex(controlPolygon, EPS);
-
-  // If the query point is at either endpoint, use direct formula
-  //   Can only use formula if the control polygon is convex
-  if(isConvexControlPolygon &&
-     ((squared_distance(q, c[0]) <= edge_tol * edge_tol) ||
-      (squared_distance(q, c[ord]) <= edge_tol * edge_tol)))
-    return convex_endpoint_winding_number(q, c, edge_tol, EPS);
+  else  // Formulas for winding number only work if shape is convex
+  {
+    // If q is outside the control polygon, for an open Bezier curve, the winding
+    //  number for the shape connected at the endpoints with straight lines is zero.
+    //  We then subtract the contribution of this line segment.
+    if(!in_polygon(q, controlPolygon, true, false, EPS))
+      return 0.0 - linear_winding_number(q, c[ord], c[0], edge_tol);
```

Comment:
Gut check, based on pattern matching with the changes above:

Should this be negated? E.g.:
```suggestion
    if(!in_polygon(q, controlPolygon, true, false, EPS))
      return linear_winding_number(q, c[ord], c[0], edge_tol);
```

I think the current way is correct since you reversed the order of `c[0]` and `c[ord]` compared to the one above (and thus reversed the orientation).

Reply:
I am very confident that the updated version is correct, as the minus sign in `linear_winding_number` was removed, which would flip the sign, but reversing the order of `c[0]` and `c[ord]` flips the sign again, keeping the behavior exactly as it was before (plus, it shouldn't pass the tests if we get rid of that minus sign as in your suggestion).

As it is, `0.0 - linear_winding_number(q, c[ord], c[0], edge_tol)` returns exactly the same number as `linear_winding_number(q, c[0], c[ord], edge_tol)`, but conceptually I believe writing it this second way is more clear. We should have that `winding_number + closure_winding_number = 0` for these exterior points, and the closure in this case is a line that connects the terminating endpoint of the Bezier curve to it's initial endpoint.

Reply:
Thanks -- I agree.